apps/opik-backend/src/main/java/com/comet/opik/domain/ExperimentDAO.java

```diff
@@ -161,8 +162,9 @@ LEFT JOIN (
                 SELECT
                     *
                 FROM experiments
-                WHERE id = :id
-                AND workspace_id = :workspace_id
+                WHERE workspace_id = :workspace_id
+                <if(id)> AND id = :id <endif>
+                <if(name)> AND name = :name <endif>
```

Comment:
Minor: please ensure either id or name is always provided in all code paths, otherwise it won't work as expected.

It seems to be like that in the code below.

Reply:
There are only 2 places where it's used. And each one adds corresponding clause, so it should be fine.